Tests/BackendIntegrationTests/StoreKitObserverModeIntegrationTests.swift

```diff
@@ -51,25 +51,34 @@ class StoreKit2ObserverModeIntegrationTests: StoreKit1ObserverModeIntegrationTes
 
         try self.testSession.disableAutoRenewForTransaction(identifier: UInt(transaction.id))
 
-        _ = try await Purchases.shared.handleObserverModeTransaction(result)
+        try await simulateAppDidBecomeActive()
+        try await Task.sleep(nanoseconds: 3_000_000_000)    // 3 sec
```

Comment:
Can we maybe observe customer info changes instead of sleeping here? Feels like this could potentially make the test flaky?

Reply:
Absolutely, I've modified it to work without the sleep! 💪